cypress/full/typeahead_page_spec.ts

```diff
@@ -0,0 +1,556 @@
+import { TypeaheadPo } from '../support/typeahead.po';
+
+describe('Typeahead demo page test suite', () => {
+  const typeahead = new TypeaheadPo();
+
+  beforeEach(() => typeahead.navigateTo());
+
+  describe('Item template', () => {
+    const itemTemp = typeahead.exampleDemosArr.itemTemplate;
+    const formTemplate = 'Model: ';
+    const textToInput = 'co';
+    const stateForCheck = 'Colorado';
+    const checkTemplate = 'This is: "Colorado" Index: 0';
+
+    it('User scrolls to Item template sub-menu and sees typeahead input and typeahead card with "Model:"', () => {
+      typeahead.scrollToMenu('Item template');
+      typeahead.isElementVisible(itemTemp, typeahead.cardHeader);
+      typeahead.isPreviewExist(itemTemp, formTemplate);
+      typeahead.isElementVisible(itemTemp, typeahead.inputSelector);
+    });
+
+    it('When user starts to type a name of a State a drop-down with matches is shown.' +
+      'The list contains a template "This is: "_matched state_" index: _index #". ',
+      () => {
+        typeahead.clearInputAndSendKeys(itemTemp, textToInput);
+        typeahead.isElementVisible(itemTemp, typeahead.activeDropdown);
+        typeahead.isElemTextCorrect(itemTemp, typeahead.activeDropdown, checkTemplate);
+      });
+
+    it('When user clicks on any item in typeahead drop-down, then typeahead container auto-fills with a selected State',
+      () => {
+        typeahead.isSelectedItemDisplayed(itemTemp, stateForCheck);
```

Comment:
typeahead.clearINP.sendKeys

Reply:
typeahead.clickOnItem

Reply:
typeahead.isSelectedItemDisplayed

Reply:
typeahead.clearInputAndSendKeys(itemTemp, textToInput);
typeahead.clickByText(this.activeDropdown, inputtedText);
typeahead.isSelectedItemDisplayed